locking/locks.go

```diff
@@ -22,74 +22,88 @@ var (
 	ErrLockAmbiguous = errors.New("lfs: multiple locks found; ambiguous")
 )
 
-// LockFile attempts to lock a file on the given remote name
+// Client is the main interface object for the locking package
+type Client struct {
```

Comment:
Exposing client objects is pretty common in API libraries. I definitely do not want to keep a global map of config objects to internal client objects.